fix/amazon-q mcp integration

harleylrn opened the pull request:
## Description
This PR improves the Amazon Q module's MCP (Model Context Protocol) integration by using the official q mcp add command instead of manually creating JSON configuration files.

## Type of Change

- [ ] New module
- [x] Bug fix
- [ ] Feature/enhancement
- [ ] Documentation
- [ ] Other

## Module Information

**Path:** `registry/coder/modules/amazon-q`  
**New version:** `v1.1.1`  
**Breaking change:** [ ] Yes [X] No

## Testing & Validation

- [x] Tests pass (`bun test`)
- [x] Code formatted (`bun run fmt`)
- [x] Changes tested locally

✅ All module tests pass:
• (pass) amazon-q module > required variables
• (pass) amazon-q module > creates coder_script resource
• (pass) amazon-q module > creates coder_app resource
✅ Terraform validation: terraform validate confirms configuration is syntactically correct
✅ Code formatting: Applied terraform fmt successfully

## Related Issues
None

## Changes Made

### Before
• Used manual JSON file creation: mkdir -p ~/.aws/amazonq and wrote to ~/.aws/amazonq/mcp.json
• Post-install script ran before MCP configuration
• Required complex JSON generation and base64 encoding

### After
• Uses official Amazon Q CLI command: q mcp add coder --command "coder" --args "exp,mcp,server,--allowed-tools,coder_report_task" --env "CODER_MCP_APP_STATUS_SLUG=amazon-q"
• MCP configuration runs before post-install scripts (proper execution order)
• Removed unused mcp_json and encoded_mcp_json local variables
• Cleaner, more maintainable code

DevelopmentCats commented:
LGTM

Have you tested this by chance just to make sure everything still works as expected?

harleylrn commented:
As I didn't change anything in the module except the order of the command execution it shouldn't affect the module in any way.
I'll try to point our coder to my repo to verify that.

DevelopmentCats commented:
> As I didn't change anything in the module except the order of the command execution it shouldn't affect the module in any way. I'll try to point our coder to my repo to verify that.

I'm pretty sure you are right too. It's very likely that it would still work, I am just always paranoid about checking these before pushing.

harleylrn commented:
> 
> I'm pretty sure you are right too. It's very likely that it would still work. I am just always paranoid about checking these before pushing.


Totally understand and you were absolutely right 😄 
Fixed the issues, now running smoothly:
```sh
Setting up Amazon Q CLI configuration...
Installing Amazon Q...
Downloading Amazon Q for x86_64...
Archive:  q.zip
  creating: q/
  inflating: q/install.sh            
  inflating: q/README                
  inflating: q/BUILD-INFO            
  creating: q/bin/
  inflating: q/bin/q                 
  inflating: q/bin/qterm             
  inflating: q/bin/qchat             
Installed Amazon Q version: q 1.12.7
Extracting auth tarball...
Extracted auth tarball
Configuring Amazon Q to report tasks via Coder MCP...

To learn more about MCP safety, see https://docs.aws.amazon.com/amazonq/latest/qdeveloper-ug/command-line-mcp-security.html


✓ Added MCP server 'coder' to 🌍 global

Added Coder MCP server to Amazon Q configuration
Running post-install script...
Running Amazon Q pre-install script logic...
Amazon Q CLI detected. Setting up rules...
Creating Amazon Q global context configuration...
Amazon Q global context configuration created at /home/coder/.aws/amazonq/global_context.json

📁 Created MCP config in '/home/coder/.amazonq/mcp.json'

To learn more about MCP safety, see https://docs.aws.amazon.com/amazonq/latest/qdeveloper-ug/command-line-mcp-security.html


✓ Added MCP server 'github_remote' to 📄 workspace

Running Amazon Q in the background with tmux...
```

DevelopmentCats commented:
LGTM I tested this and it loads up the coder mcp server as expected and tasks run as expected.

@harleylrn can you bump the version to 1.1.2 since this is a patch, and then I can get this merged

DevelopmentCats commented:
@hugodutka can we get your final approval on this?
